### ISSUE TITLE
a11y: Remove right-click context menu from tables

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#14154]
Remove right-click context menus from tables used throughout Rancher Dashboard. The decision came from discussions that concluded these context menus are not discoverable or accessible for users.

## Proposed Changes

- Remove right-click context menus from tables
- Ensure that all context menus are available through more discoverable means (e.g., toolbars, buttons, or keyboard shortcuts)

## Acceptance Criteria

- [ ] Right-click context menus are no longer present in tables
- [ ] Previously available context menus are accessible through alternative means

## Details

Right-click context menus were initially implemented to provide quick access to table-related actions. Review of the context menus indicates that these menus are not intuitive and are often overlooked. This lack of discoverability impacts user experience, especially for users who rely on keyboard navigation or screen readers.

**Comment by yonasberhe23** [comment]
Requires manual validation

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.12-d6ed163883fa80bb91f3ad4320c494e6c5e1860a-head
UI:master c4bf19e

I noticed when the table is empty we still showing the menu for right click 🔴 

https://github.com/user-attachments/assets/de92805c-da58-4d2a-837d-bd65b418e5f4

**Comment by rak-phillip** [comment]
@a-arias In the recording you shared, it looks like the offending right-click menus are coming from the browser. This PR removed the custom, table specific, menus that Rancher provided. Can you test again to confirm that we don't see menus with actions like Execute Shell, View Logs, Edit Config, etc..?

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.12-719f9ed0c7bfc2e2ba9213e151256feeb7d095b5-head
UI: master https://github.com/rancher/dashboard/commit/ce49fa87705d0975649ada5ee2f3f14aec1a8dd8


menus removed from tables ✅ 

https://github.com/user-attachments/assets/5fdd2bc9-26ff-4f4f-b632-ea822edce4b4

https://github.com/user-attachments/assets/5c9075dd-4fd6-4c47-9315-954ee8bb3ab9

https://github.com/user-attachments/assets/be1e7551-1cc2-45e4-9a94-dc84f6b34369

https://github.com/user-attachments/assets/b9043615-ff89-4aa0-822c-b83086286a1a
